### PR TITLE
TypeScript types for sidecar CLI args

### PR DESCRIPTION
**Pull request by danvk**
Follow-on to #126 to generate TypeScript types for the CLI args for the sidecar (this temporary solution has had more staying power than we expected!).

The flags are only specified via interactions with `ArgumentParser`. This has to go through some internal APIs and make some assumptions about what's required/optional to work.

An alternative approach would be to make all the subcommands take a single JSON argument and then we could use Pydantic to model that, same as we do with outputs. This would be a lot simpler/cleaner, but it would make sidecar invocations from the command line less idiomatic, i.e.

    python cli.py ingest '{"pdf_directory": "path/to/pdfs"}'

vs the current:

    python cli.py ingest --pdf_directory path/to/pdfs

Would love some input from @gjreda on whether this would be annoying or if we should just embrace JSON.

### CODE REVIEW

File: src/api/__tests__/chat.test.ts
Comment (danvk) [inline src/api/__tests__/chat.test.ts:4]: `vi.mock('module')` mocks all the functions in the module. We only want it to mock one (`callSidecarWithArgs`, not `callSidecar`). The easiest way is to mock just one function in a module is to just move it into a standalone module. https://stackoverflow.com/q/59312671/388951

File: python/generate_schema.py
Comment (danvk) [inline python/generate_schema.py:48]: I don't love all this code. See my idea in the PR description about making the CLI take JSON instead.

File: src/api/__tests__/chat.test.ts
Comment (danvk) [reply src/api/__tests__/chat.test.ts]: (this is irrelevant after the refactor to use Pydantic)

File: python/generate_schema.py
Comment (danvk) [reply python/generate_schema.py]: This gross code is gone!

File: python/README.md
Comment (danvk) [inline python/README.md:291]: Looks like `update` had already hit on the one-arg JSON form :)

**Review by gjreda — APPROVED** [review]
LGTM, but let's hold off on merging until next week since merging this will cause some conflicts to other PRs. Once the other PRs get merged, I will resolve the merge conflicts on this one.

**Review by cguedes — APPROVED** [review]

**Comment by cguedes** [comment]
> LGTM, but let's hold off on merging until next week since merging this will cause some conflicts to other PRs. Once the other PRs get merged, I will resolve the merge conflicts on this one.

Ok. Will make this a draft PR.

**Comment by gjreda** [comment]
Resolved the expected merge conflicts. For the sidecar, I'm ok with us merging this now. What do you think @cguedes?

**Review by gjreda — APPROVED** [review]

**Comment by cguedes** [comment]
@gjreda you can merge it 💪

**Pull request merged**
